P/pocl/bundled/patches/sysroot.patch

```diff
@@ -0,0 +1,10 @@
+--- a/cmake/bitcode_rules.cmake
++++ b/cmake/bitcode_rules.cmake
+@@ -57,6 +57,7 @@
+         ${KERNEL_C_FLAGS} "-o" "${BC_FILE}" "-c" "${FULL_F_PATH}"
+         "-I${CMAKE_SOURCE_DIR}/include"
+         "-include" "${CMAKE_SOURCE_DIR}/include/_kernel_c.h"
++        "-I/opt/x86_64-linux-musl/x86_64-linux-musl/sys-root/usr/include/"
```

Comment:
Where could I find the target's `math.h` instead of the host's?

Reply:
It should be in `/opt/${target}/${target}/sys-root/usr/include` (what you used is `/opt/${MACHTYPE}/${MACHTYPE}/sys-root/usr/include`)

Reply:
Great, thanks. That does seem to break the Apple build though:

```
/opt/aarch64-apple-darwin20/aarch64-apple-darwin20/sys-root/usr/include/stdio.h:220:5: error: 'TARGET_OS_IPHONE' is not defined, evaluates to 0 [-Werror,-Wundef-prefix=TARGET_OS_]
#if TARGET_OS_IPHONE
    ^
```

IIUC, TARGET_OS_IPHONE etc are set by `TargetConditionals.h` which should be included automatically... if the directory is included via `isysroot` instead? Doing so makes it not find `math.h` anymore though. Also, AFAICT the CMake toolchain file should already make sure the sysroot is set correctly.

Reply:
What if you remove the `-Werror` flag? :grimacing:

Reply:
Worked around this by inserting an include for `TargetConditionals.h`, but happy to hear if you have other ideas.

Reply:
> What if you remove the `-Werror` flag? 😬

That will be tricky, as it's [inserted by Clang](https://github.com/llvm/llvm-project/blob/ef7d46c7e6b33b6294f23b6df290b335a761fbd5/clang/lib/Driver/ToolChains/Darwin.cpp#L1147-L1150)...